image_search_engine/src/data/loader.py

```diff
@@ -0,0 +1,53 @@
+import torch
+import cv2
+import albumentations as A
+from image_search_engine.src import config as CFG
+
+class CLIPLoader(torch.utils.data.Dataset):
+    def __init__(self, image_filenames, captions, tokenizer, transforms):
```

Comment:
Since Python is a dynamically typed language, stating the types of parameters is not necessary. However, the pythonic way would be to use type hints which would allow both the reader and linter (and mypy) to give you suggestions while writing code.